src/Metadata/Property/Factory/XmlPropertyMetadataFactory.php

```diff
@@ -0,0 +1,194 @@
+        if ($this->decorated) {
+            try {
+                $parentPropertyMetadata = $this->decorated->create($resourceClass, $property, $options);
+            } catch (PropertyNotFoundException $propertyNotFoundException) {
+                // Ignore not found exception from decorated factories
+            }
+        }
+
+        if (!class_exists($resourceClass) || !property_exists($resourceClass, $property) || empty($propertyMetadata = $this->getMetadata($resourceClass, $property))) {
+            return $this->handleNotFound($parentPropertyMetadata, $resourceClass, $property);
+        }
+
+        if ($parentPropertyMetadata) {
+            return $this->update($parentPropertyMetadata, $propertyMetadata);
+        }
+
+        return new PropertyMetadata(
+            null,
+            $propertyMetadata['description'],
+            $propertyMetadata['readable'],
+            $propertyMetadata['writable'],
+            $propertyMetadata['readableLink'],
+            $propertyMetadata['writableLink'],
+            $propertyMetadata['required'],
+            $propertyMetadata['identifier'],
+            $propertyMetadata['iri'],
+            null,
+            $propertyMetadata['attributes']
+        );
+    }
+
+    /**
+     * Returns the metadata from the decorated factory if available or throws an exception.
+     *
+     * @param PropertyMetadata|null $parentPropertyMetadata
+     * @param string                $resourceClass
+     * @param string                $property
+     *
+     * @throws PropertyNotFoundException
+     *
+     * @return PropertyMetadata
+     */
+    private function handleNotFound(PropertyMetadata $parentPropertyMetadata = null, string $resourceClass, string $property) : PropertyMetadata
+    {
+        if ($parentPropertyMetadata) {
+            return $parentPropertyMetadata;
+        }
+
+        throw new PropertyNotFoundException(sprintf('Property "%s" of the resource class "%s" not found.', $property, $resourceClass));
+    }
+
+    /**
+     * Extracts metadata from the XML tree.
+     *
+     * @param string $resourceClass
+     * @param string $propertyName
+     *
+     * @throws InvalidArgumentException
+     *
+     * @return array
+     */
+    private function getMetadata(string $resourceClass, string $propertyName) : array
+    {
+        foreach ($this->paths as $path) {
+            try {
+                $domDocument = XmlUtils::loadFile($path, self::RESOURCE_SCHEMA);
+            } catch (\InvalidArgumentException $e) {
+                throw new InvalidArgumentException($e->getMessage(), $e->getCode(), $e);
+            }
+
+            $properties = (new \DOMXPath($domDocument))->query(sprintf('//resources/resource[@class="%s"]/property[@name="%s"]', $resourceClass, $propertyName));
+
+            if (false === $properties || 0 >= $properties->length || null === $properties->item(0) || false === $property = simplexml_import_dom($properties->item(0))) {
```

Comment:
Could you change this for the sake of lisibility ?
`if ( blabla
   || 
     blabla)`